fw/http_msg.c

```diff
@@ -553,13 +553,20 @@ tfw_http_msg_hdr_close(TfwHttpMsg *hm)
 	TfwHttpHdrTbl *ht = hm->h_tbl;
 	TfwHttpParser *parser = &hm->stream->parser;
 	unsigned int id = parser->_hdr_tag;
+	bool is_srv_conn = TFW_CONN_TYPE(hm->conn) & Conn_Srv;
 
 	BUG_ON(parser->hdr.flags & TFW_STR_DUPLICATE);
 	BUG_ON(id > TFW_HTTP_HDR_RAW);
 
 	/* Close just parsed header. */
 	parser->hdr.flags |= TFW_STR_COMPLETE;
 
+	if (!is_srv_conn &&
+	    (!tfw_http_check_header_size(hm) ||
+	     !tfw_http_check_headers_count(hm) ||
+	     !tfw_http_check_header_list_size(hm)))
```

Comment:
Good question. Probably we can just process the `-EPREM` return value in the caller and go with the security violation path? Let's discuss this on the meeting

Reply:
After our private discussion we decide to leave it here because it is the earliest place where we can determine that this limits are exceeded. Add comment